File: src/zhengkuan/yzk/EditorInject.java
//package zhengkuan.yzk;
//
//import com.intellij.openapi.fileEditor.FileEditorManager;
//import com.intellij.openapi.fileEditor.FileEditorManagerListener;
//import com.intellij.openapi.project.Project;
//import com.intellij.openapi.vfs.VirtualFile;
//import org.jetbrains.annotations.NotNull;
//
///**
// * @author zhengkuan.yzk@alibaba-inc.com
// * @date 2018/7/8
// */
//public class EditorInject implements FileEditorManagerListener {
//
//    private Project project;
//
//    public EditorInject(Project project) {
//        this.project = project;
//    }
//
//    @Override
//    public void fileOpened(@NotNull FileEditorManager source, @NotNull VirtualFile file) {
//        System.out.println(file.getName());
//    }
//}

File: src/inject/EditorInject.kt
package inject

import com.intellij.openapi.fileEditor.FileEditorManager
import com.intellij.openapi.fileEditor.FileEditorManagerEvent
import com.intellij.openapi.fileEditor.FileEditorManagerListener
import com.intellij.openapi.project.Project
import com.intellij.openapi.vfs.VirtualFile
import zhengkuan.yzk.AutoTransformEncodingAction


class EditorInject(private var project: Project) : FileEditorManagerListener {
    override fun selectionChanged(event: FileEditorManagerEvent) {

        var source = event.manager
        var file = event.newFile
        print("\n" + file!!.name + " ")

        AutoTransformEncodingAction.transformFileEncoding(file)
    }

    override fun fileOpened(source: FileEditorManager, file: VirtualFile) {
    }

    override fun fileClosed(source: FileEditorManager, file: VirtualFile) {
    }
}